packages/yourloops/test/integration/assert/device-page.assert.ts

```diff
@@ -0,0 +1,56 @@
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import { screen, within } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+
+export const checkDeviceSettingsContent = () => {
+  const deviceSettings = screen.getByTestId('device-settings-container')
+  expect(deviceSettings).toHaveTextContent('Last upload: 1/1/2020Copy as textDBLDBLG1ManufacturerDiabeloopIdentifier1234IMEI1234567890Software version1.0.5.25PumpManufacturerVICENTRASerial number123456Pump versionbetaCartridge expirationApr 12, 2050CGMManufacturerDexcomProductG6Sensor expirationApr 12, 2050Transmitter software versionv1Transmitter IDa1234Transmitter expirationApr 12, 2050SettingsSettingsSettingValueUnitWEIGHT72.0kgSettings HistoryLevelSettingValueDateNov 7, 2022 3:01 pm1BOLUS_AGGRESSIVENESS_FACTOR143 %Nov 1, 2022 1:00 am1LARGE_MEAL_BREAKFAST150.0 gNov 1, 2022 1:00 am1LARGE_MEAL_DINNER150.0 gNov 1, 2022 1:00 am1LARGE_MEAL_LUNCH70.0 gNov 1, 2022 1:00 am1MEAL_RATIO_BREAKFAST_FACTOR100 %110 %Nov 2, 2022 6:00 pm1MEAL_RATIO_BREAKFAST_FACTOR100 %100 %Nov 1, 2022 1:00 am1MEAL_RATIO_BREAKFAST_FACTOR100 %Nov 1, 2022 1:00 am1MEAL_RATIO_DINNER_FACTOR100 %90 %Nov 2, 2022 6:00 pm1MEAL_RATIO_DINNER_FACTOR100 %100 %Nov 1, 2022 1:00 am1MEAL_RATIO_DINNER_FACTOR100 %Nov 1, 2022 1:00 am1MEAL_RATIO_LUNCH_FACTOR130 %90 %Nov 7, 2022 3:01 pm1MEAL_RATIO_LUNCH_FACTOR130 %Nov 1, 2022 1:00 am1MEDIUM_MEAL_BREAKFAST70.0 gNov 1, 2022 1:00 am1MEDIUM_MEAL_DINNER60.0 gNov 1, 2022 1:00 am1MEDIUM_MEAL_LUNCH50.0 gNov 1, 2022 1:00 am1PATIENT_BASAL_AGGRESSIVENESS_FACTOR_LEVEL_IN_EUGLYCAEMIA100 %Nov 1, 2022 1:00 am1PATIENT_GLY_HYPER_LIMIT180.1 mg/dL140.0 mg/dLNov 2, 2022 8:00 am1PATIENT_GLY_HYPER_LIMIT180.1 mg/dL180.1 mg/dLNov 1, 2022 1:00 am1PATIENT_GLY_HYPER_LIMIT180.1 mg/dLNov 1, 2022 1:00 am1PATIENT_GLY_HYPO_LIMIT70.0 mg/dL60.0 mg/dLNov 2, 2022 8:00 am1PATIENT_GLY_HYPO_LIMIT70.0 mg/dL70.0 mg/dLNov 1, 2022 1:00 am1PATIENT_GLY_HYPO_LIMIT70.0 mg/dLNov 1, 2022 1:00 am1PATIENT_GLYCEMIA_TARGET100.0 mg/dLNov 1, 2022 1:00 am1SMALL_MEAL_BREAKFAST15.0 gNov 1, 2022 1:00 am1SMALL_MEAL_DINNER20.0 gNov 1, 2022 1:00 am1SMALL_MEAL_LUNCH30.0 gNov 1, 2022 1:00 am1TOTAL_INSULIN_FOR_24H53.0 UNov 1, 2022 1:00 am1WEIGHT69.0 kgNov 1, 2022 1:00 am')
+}
+
+export const checkCopyTextButton = async () => {
+  const writeText = jest.fn()
+  Object.assign(navigator, {
+    clipboard: {
+      writeText
+    }
+  })
+  const deviceSettings = screen.getByTestId('device-settings-container')
+  const copyTextButton = within(deviceSettings).getByRole('button', { name: 'Copy as text' })
+  await userEvent.click(copyTextButton)
+
+  expect(navigator.clipboard.writeText).toHaveBeenCalledTimes(1)
```

Comment:
```suggestion
  const copiedStringToPaste = '01/01/2020, 00:00:00\n\n-- DBL --\nManufacturer      Diabeloop\nIdentifier        1234\nIMEI              1234567890\nSoftware version  1.0.5.25\n\n-- Settings --\nName    Value  Unit\nWEIGHT   72.0  kg'
  expect(navigator.clipboard.writeText).toHaveBeenCalledWith(copiedStringToPaste)
```

:warning: The date I gave probably does not work on your machine.
A lesser solution (to having the date working on every machine) could:
```
const copiedStringToPaste = '/2020, 00:00:00\n\n-- DBL --\nManufacturer      Diabeloop\nIdentifier        1234\nIMEI              1234567890\nSoftware version  1.0.5.25\n\n-- Settings --\nName    Value  Unit\nWEIGHT   72.0  kg'
expect(navigator.clipboard.writeText).toHaveBeenCalledWith(expect.stringContaining(copiedStringToPaste))
```